ui/src/main/java/org/pentaho/di/ui/core/widget/TableView.java

```diff
@@ -626,6 +630,18 @@ public void run() {
         }
         tableViewModifyListener.cellFocusLost( rownr );
       }
+
+      /**
+       * This is a workaround for SWT bug (see PDI-15268).
+       * Calling a context menu should be ignored in
+       * SWT org.eclipse.swt.widgets.Control#gtk_event_after
+       *
+       * @return true if it is wrong event
+       */
+      private boolean isWrongLostFocusEvent() {
+        Control controlGotFocus = Display.getCurrent().getCursorControl();
+        return Const.isLinux() && ( controlGotFocus == null || text.equals( controlGotFocus ) );
```

Comment:
Should there not be a null check on the text variable ?
return Const.isLinux() && ( controlGotFocus == null || ( text != null && text.equals( controlGotFocus ) ) ;

Reply:
@mbatchelor 
It's an event for a text Widget "text". It will be despose in line 599 ftext.dispose();. Before my changes we don't have NPE in L599.